components/tooltip/Tooltip.js

```diff
@@ -28,203 +28,201 @@ const SIZES = {
   },
 };
 
-const tooltipFactory = () => {
-  return ComposedComponent => {
-    class TooltippedComponent extends Component {
-      static propTypes = {
-        children: PropTypes.node,
-        className: PropTypes.string,
-        onClick: PropTypes.func,
-        onMouseEnter: PropTypes.func,
-        onMouseLeave: PropTypes.func,
-        tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-        tooltipColor: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'inverse']),
-        tooltipHideOnClick: PropTypes.bool,
-        tooltipIcon: PropTypes.element,
-        tooltipPosition: PropTypes.oneOf(Object.keys(POSITIONS).map(key => POSITIONS[key])),
-        tooltipShowOnClick: PropTypes.bool,
-        tooltipSize: PropTypes.oneOf(Object.keys(SIZES)),
-        documentObject: PropTypes.object.isRequired,
+const Tooltip = ComposedComponent => {
+  class TooltippedComponent extends Component {
+    static propTypes = {
+      children: PropTypes.node,
+      className: PropTypes.string,
+      onClick: PropTypes.func,
+      onMouseEnter: PropTypes.func,
+      onMouseLeave: PropTypes.func,
+      tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+      tooltipColor: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'inverse']),
+      tooltipHideOnClick: PropTypes.bool,
+      tooltipIcon: PropTypes.element,
+      tooltipPosition: PropTypes.oneOf(Object.keys(POSITIONS).map(key => POSITIONS[key])),
```

Comment:
PP: I think a quicker way of doing the same thing would be doing `Object.values(POSITIONS)`